packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3012,20 +3059,194 @@ class InheritedElement extends ProxyElement {
+        'This will happen if an InheritedWidgetLinkParent is linked to an '
+        'InheritedWidgetLinkChild ancestor.'
+      );
+    }
+    return true;
+  }
+
+  @override
+  void activate() {
+    super.activate(); // clears _dependencies, and sets active to true
+    assert(_debugLinkIsValid);
+    if (link != null && link._active)
+      link._markLinkChildNeedsBuild();
+  }
+
+  @override
+  void deactivate() {
+    super.deactivate();
+    assert(_debugLinkIsValid);
+    if (link != null && link._active)
+      link._clearDependenciesRecursively(link);
+  }
+
+  @override
+  void _updateInheritance() {
+    super._updateInheritance();
+    assert(_debugLinkIsValid());
+    if (link != null && link._active) {
+      assert(_debugLinkIsNotCyclic);
+      link._updateInheritanceRecursively(link);
+    }
+  }
+
+  @override
+  void notifyClients(InheritedWidgetLinkParent oldWidget) {
+    if (!widget.updateShouldNotify(oldWidget))
+      return;
+    InheritedElementLinkChild oldLinkChild = oldWidget.link?._currentElement;
+    if (oldLinkChild != null) {
+      oldLinkChild._clearDependenciesRecursively(oldLinkChild);
+      oldLinkChild._updateInheritance();
+    }
+    assert(_debugLinkIsValid);
+    link?._updateInheritance();
+  }
+}
+
+/// An element that uses a [InheritedWidgetLinkChild] as its configuration.
+class InheritedElementLinkChild extends ProxyElement {
+  InheritedElementLinkChild(InheritedWidgetLinkChild widget) : super(widget);
+
+  @override
+  InheritedWidgetLinkChild get widget => super.widget;
+
+  InheritedElementLinkParent get link => widget.link?._currentElement;
+
+  @override
+  int get depth {
+    if (_depth == null)
+      return null;
+    return math.max(_depth, 1 + (link?.depth ?? 0));
+  }
+
+  @override
+  bool _debugIsInScope(Element target) {
+    return super._debugIsInScope(target) || (link?._debugIsInScope(target) ?? false);
+  }
+
+  bool _debugLinkIsValid() {
+    if (widget.link != null && !(widget.link is GlobalKey)) {
+      throw new FlutterError(
+        'An InheritedElementLinkChild\'s key must either by a GlobalKey or null.'
+      );
+    }
+
+    final Element linkElement = widget.link?._currentElement;
+    if (linkElement != null && !(linkElement is InheritedElementLinkParent)) {
+      throw new FlutterError(
+        'InheritedWidgetLinkChild link value is not an InheritedWidgetLinkParent.\n'
+        'The value of an InheritedWidgetLinkChild\'s link must be a GlobalKey which '
+        'was given as an InheritedWidgetLinkParent\'s key.'
+      );
+    }
+
+    return true;
+  }
+
+  void _markLinkChildNeedsBuild() {
+    assert(_debugLifecycleState != _ElementLifecycle.defunct);
+    if (!_active)
+      return;
+    assert(owner != null);
+    assert(_debugLifecycleState == _ElementLifecycle.active);
+    assert(link != null);
+    assert(() {
+      if (link.owner._debugBuilding) {
+        if (link.owner._debugCurrentBuildTarget == null) {
```

Comment:
in fact in general you should see if this function needs updating; markNeedsBuild changed a bit recently